optimize.c

```diff
@@ -2051,12 +2051,18 @@ intern_blocks(opt_state_t *opt_state, struct icode *ic)
 static void
 opt_cleanup(opt_state_t *opt_state)
 {
-	free((void *)opt_state->vnode_base);
-	free((void *)opt_state->vmap);
-	free((void *)opt_state->edges);
-	free((void *)opt_state->space);
-	free((void *)opt_state->levels);
-	free((void *)opt_state->blocks);
+	if (opt_state->vnode_base)
```

Comment:
C90 (and subsequent versions - and, I think, C89) explicitly require that, if a null pointer is passed to free(), nothing happens ("If `ptr` is a null pointer, no action occurs."), so the tests aren't a requirement, just a code style choice.  (I'm OK either way.)

Reply:
I did not know it was standardized.
If this is only code style, this should go into another pull request...